intel_extension_for_transformers/llm/runtime/graph/core/ne_layers.c

```diff
@@ -7946,6 +7955,50 @@ static void ne_compute_forward_rope_f16(const struct ne_compute_params* params,
   }
 }
 
+static void ne_compute_forward_rope_jblas(const struct ne_compute_params* params, const struct ne_tensor* src0,
+                                          const struct ne_tensor* src1, struct ne_tensor* dst) {
+  if (params->type == NE_TASK_INIT || params->type == NE_TASK_FINALIZE) return;
+  NE_ASSERT(("use internal multi-threading", params->nth == 1));
+
+  NE_ASSERT(src1->type == NE_TYPE_I32);
+  NE_ASSERT(ne_nelements(src1) == 5);  // 5 params
+
+  const int n_past = ((int32_t*)src1->data)[0];
+  const int n_dims = ((int32_t*)src1->data)[1];
+  const int mode = ((int32_t*)src1->data)[2];
+  const int prompt_size = ((int32_t*)src1->data)[3];
+  const int n_keep = ((int32_t*)src1->data)[4];
+  const bool skip = mode & 1;
+  const bool is_neox = mode & 2;
+  const bool is_glm = mode & 4;
+  NE_ASSERT(("glm mode RoPE is not implemented!", !is_glm));
```

Comment:
Chat GLM v1 repo is not implemented? 
it's different from Chat GLM v2 right ?

Reply:
ChatGLM1 and neox uses their special pattern of RoPE which are different from all other models, and I'm not sure if and how these special patterns can change the shift-RoPE computation.

ChatGLM2 uses the "common" pattern of RoPE just like most other models.